first_order_model/shrink_util.py

```diff
@@ -456,18 +485,34 @@ def add_names(names):
             'bottleneck.r5.norm1', 'bottleneck.r5.conv1',
             'bottleneck.r5.norm2', 'bottleneck.r5.conv2', 'up_blocks.0.conv',
             'up_blocks.0.norm', 'up_blocks.1.conv', 'up_blocks.1.norm',
-            'hr_up_blocks.0.conv', 'hr_up_blocks.0.norm', 'final'
+            'hr_up_blocks.0.conv', 'hr_up_blocks.0.norm'
         ])
+        if is_1024:
+            add_names(['hr_up_blocks.0.norm', 'hr_up_blocks.1.conv', 'hr_up_blocks.1.norm', 'final'])
+        else:
+            add('hr_up_blocks.0.norm', 'final')
 
         # Features get concatted into down block
         add('down_blocks.1.norm', 'bottleneck.r0.norm1')
 
-        # Second up block has 32 lr features added
-        add('lr_first.norm', 'up_blocks.0.conv')
+
+        if is_1024:
+            # Second up block has 32 lr features added
+            add('lr_first.norm', 'up_blocks.1.conv')
+        else:
+            # Second up block has 32 lr features added
+            add('lr_first.norm', 'up_blocks.0.conv')
 
         # Add 2x hr down outputs to first hr up
-        add('hr_down_blocks.0.norm', 'hr_up_blocks.0.conv')
-        add('hr_down_blocks.0.norm', 'hr_up_blocks.0.conv')
+
+        if is_1024:
+            add('hr_down_blocks.1.norm', 'hr_up_blocks.0.conv')
```

Comment:
Oh that is because they are the hr up block accepts the down block concatenated with itself.

Reply:
0->0 is just the way the indices work out for 512 since its
down_blocks = 0
up_blocks     =  0

but 1024 is 
down-blocks = 0 1
up blocks      =  0 1
so the last down block feeds to the first up block.

Reply:
Does the image help here. Resolving because there are multiple 512 vs 1024 comments.